commands/geosearch.md

```diff
@@ -1,16 +1,20 @@
-Return the members of a sorted set populated with geospatial information using `GEOADD`, which are within the borders of the area specified by a given shape. This command extends the `GEORADIUS` command, so in addition to searching within circular areas, it supports searching within rectangular areas.
+Return the members of a sorted set populated with geospatial information using `GEOADD`, which are within the borders of the area specified by a given shape. This command extends the `GEORADIUS` command, so in addition to searching within circular areas, it supports searching within rectangular areas and it supports searching within a "polygon" which is a user defined area enclosed by polygon vertices.
 
 This command should be used in place of the deprecated `GEORADIUS` and `GEORADIUSBYMEMBER` commands.
 
-The query's center point is provided by one of these mandatory options:
-
-* `FROMMEMBER`: Use the position of the given existing `<member>` in the sorted set.
-* `FROMLONLAT`: Use the given `<longitude>` and `<latitude>` position.
-
 The query's shape is provided by one of these mandatory options:
 
 * `BYRADIUS`: Similar to `GEORADIUS`, search inside circular area according to given `<radius>`.
 * `BYBOX`: Search inside an axis-aligned rectangle, determined by `<height>` and `<width>`.
+* `BYPOLYGON`: Search within an area enclosed by the polygon vertices, determined by `<num-vertices> <lon> <lat> [<lon> <lat>  ...]`.
+
+For `BYBOX` and `BYRADIUS` based queries, the center point is provided by one of these mandatory options:
+
+* `FROMMEMBER`: Use the position of the given existing `<member>` in the sorted set.
+* `FROMLONLAT`: Use the given `<longitude>` and `<latitude>` position.
+
+Note: In case of `BYPOLYGON` based queries, the center point and the bounding box are computed based on the polygon vertices of the command.
+With this option, providing `FROMMEMBER` or `FROMLONLAT` is invalid.
```

Comment:
This is not a side note. It's as important as the fact that these are mandatory for BYBOX and BYRADIUS. We can skip `Note: `.

```suggestion
In case of `BYPOLYGON` based queries, the center point and the bounding box are computed based on the polygon vertices of the command.
With this option, providing `FROMMEMBER` or `FROMLONLAT` is invalid.
```